internal/pkg/runtime/launcher/oci/launcher_linux.go

```diff
@@ -54,10 +55,10 @@ func checkOpts(lo launcher.Options) error {
 	if lo.WritableTmpfs {
 		badOpt = append(badOpt, "WritableTmpfs")
 	}
-	if lo.OverlayPaths != nil {
+	if lo.OverlayPaths != nil && len(lo.OverlayPaths) > 0 {
```

Comment:
Redundant check here (and below in a few places.) Can simplify to:

```suggestion
	if len(lo.OverlayPaths) > 0 {
```